vars/utils.groovy

```diff
@@ -520,14 +523,25 @@ def pytestSuite3() {
         env.gateway_user = "${gateway_user}"
         env.PYTHONPATH=env.WORKSPACE
         env.PATH="${env.WORKSPACE}/tuna:${env.PATH}"
-        //addMachine(arch, num_cu)
-        // download the latest perf db
-        //runsql("DELETE FROM config_tags; DELETE FROM job; DELETE FROM config;")
         sshagent (credentials: ['bastion-ssh-key']) {                 
-           // test fin builder and test fin builder conv in sequence
-           sh "pytest tests/test_fin_evaluator.py -s"
-           sh "pytest tests/test_update_golden.py -s"
+           sh "python3 -m coverage run -a -m pytest tests/test_fin_evaluator.py -s"
+           sh "python3 -m coverage run -a -m pytest tests/test_update_golden.py -s"
+           sh "coverage report -m"
+           sh "python3 -m coverage json"
+        }
+        if (current_run == main_branch) {
+            sh "python3 tests/covscripts/coverage.py ${main_branch}"
+            archiveArtifacts artifacts: "${env.COVERAGE_ARTIFACT_FILE_NAME}", allowEmptyArchive: true, fingerprint: true
+        } else {
+        try {
+            sh "wget ${env.TUNA_COVERAGE_URL}/${main_branch}/lastSuccessfulBuild/artifact/${env.COVERAGE_ARTIFACT_FILE_NAME}"
+            sh "python3 tests/covscripts/coverage.py ${current_run}"
+        } catch (Exception err) {
+            currentBuild.result = 'SUCCESS'
```

Comment:
@reidkwja when does this fail? and does it fail if the coverage decreased?

Reply:
Changed back...was there for testing the branch